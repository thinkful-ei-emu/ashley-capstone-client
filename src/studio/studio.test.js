import React from 'react';
import Studio from './studio';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe(`Studio component`, () => {
  const props = {
    artwork: [
      {
        id: 1,
        title: 'Dogs',
        uploaded: '2019-01-03T00:00:00.000Z',
        gallery_id: 1,
        artpiece_image: 'placeholder.png'
      },
      {
        id: '2',
        title: 'Cats',
        uploaded: '2019-01-03T00:00:00.000Z',
        gallery_id: '2',
        artpiece_image: 'placeholder.png'
      },
      {
        id: '3',
        title: 'Silly Cats',
        uploaded: '2019-01-03T00:00:00.000Z',
        gallery_id: '2',
        artpiece_image: 'placeholder.png'
      }
    ],
    galleries: [
      {
        id: '1',
        name: 'Dogs'
      },
      {
        id: '2',
        name: 'Cats'
      },
      {
        id: '3',
        name: 'Birds'
      }
    ]
  };

  it('renders empty list', () => {
    //find array of children and look at length
    const wrapper = shallow(<Studio galleries={[]} />);
    expect(wrapper.find('art-gallery-select').children().length).toEqual(0);
  });

  it('renders a gallery option for each gallery in array', () => {
    const options = shallow(<Studio {...props} />).find('art-gallery-select');
    expect(toJson(options)).toMatchSnapshot();
  });
});
