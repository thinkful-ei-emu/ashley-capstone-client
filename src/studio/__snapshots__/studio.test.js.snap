// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Studio component renders a gallery option for each gallery in array 1`] = `null`;
